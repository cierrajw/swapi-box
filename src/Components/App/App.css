@font-face {
  font-family: starWars;
  src: url(../../assets/fonts/Starjhol.ttf)
}

body{
  background-image: url('../../assets/Images/maxresdefault.jpg');
  background-repeat: no-repeat;
  background-position: center;
}

/*.swapi-button-section{
  display: flex;
  flex-direction: column;
}*/

/*.intro-page {
  margin: 0 auto;
}*/

.intro-page {
  margin: 0 auto;
  display: block;
}

.main-div{
  /*border: 5px groove #106282;*/
  border-radius: 5px;
  margin: 6em;
  width: 70rem;
  height: 100vh;
  margin: 8rem auto;
}

.swapi-intro-title{
  font-family: starWars;
  padding: .4rem;
  letter-spacing: .8rem;
  margin:auto;
  color: #F2B705;
  font-size: 5.5rem;
  text-align: center;
}

.intro-swapi-button{
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-left: 47%;
/*  background-color: #73000A;
  clip-path: polygon(50% 0%, 61% 35%, 98% 35%, 68% 57%, 79% 91%, 50% 70%, 21% 91%, 32% 57%, 2% 35%, 39% 35%);*/
  /*margin: 0 auto;*/
  /*background-color: #F2B705;*/
  /*color: black;*/
  background-color: black;
}

.crawl-text-div{
   position:absolute;
   top:0;
   bottom:0;
   left:0;
   right:0;
   transform-origin:50% 100%;
   transform:perspective(300px) rotateX(20deg);
}

.film-text{
   color: #F2B705;
   position:absolute;
   top:-3500px;
   bottom:0;
   left:0;
   right:0;
   overflow:hidden;
   text-align:center;
   font-family:sans-serif;
   font-size: 3rem;
   padding-top:3500px;
   animation:autoscroll 2000s linear;
}

.no-display{
  display: none;
}

@keyframes autoscroll{
    to{
        margin-top:-50000px;
    }
}

@keyframes swapi-intro-title {
  0% {
    opacity: 0;
    transform: scale(1.5) translateY(-0.75em);
  }
  20% {
    opacity: 1;
  }
  89% {
    opacity: 1;
    transform: scale(1);
  }
  100% {
    opacity: 0;
    transform: translateZ(-1000em);
  }
}

@media screen and (max-width: 1113px){
  
}
