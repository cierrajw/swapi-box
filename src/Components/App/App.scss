.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.component-container{
  display: grid;
  grid-template-columns: 1fr 1fr;
  height: 100vh;
}

.right-section{
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 3fr;
}

.scroll-section{
  grid-column: 1;
  grid-row: 1/3;
  height: 100vh;
}

.header{
  background-color: yellow;
  grid-column: 2;
  grid-row: 1;
}

.card-container{
  grid-row: 2;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
