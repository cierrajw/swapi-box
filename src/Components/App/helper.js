export const fetchPeople = async () => {
 try {
   const url = 'https://swapi.co/api/people/';
   const response = await fetch(url);
   const data = await response.json();
   const unresolvedCharacterPromises = data.results.map(async character => {
     const person_name = character.name
     const species = await fetchSpecies(character.species)
     const homeworld = await fetchHomeWorld(character.homeworld)
     const peopleCard = {person_name, ...species, ...homeworld, type: 'people', id: Date.now() * Math.random()}
     return peopleCard
   })
   return Promise.all(unresolvedCharacterPromises)
 } catch(error) {
   throw new Error(error.message);
 }
}

export const fetchSpecies = async(url) => {
 try {
   const response = await fetch(url);
   const speciesData = await response.json();
   return {species: speciesData.name, language: speciesData.language, type: 'species'}
 } catch(error) {
   throw new Error(error.message)
 }
}

export const fetchHomeWorld = async (url) => {
 try {
   const response = await fetch(url);
   const homeworldData = await response.json();
   return {homeworld: homeworldData.name, population: homeworldData.population}
 } catch(error) {
   throw new Error(error.message);
 }
}

export const fetchVehicles = async () =>{
  try{
    const url = "https://swapi.co/api/vehicles/";
    const response = await fetch(url);
    const data = await response.json();

  const unresolvedVehiclePromises = data.results.map(async vehicle =>{
    const vehicle_name = vehicle.name;
    const model = vehicle.model;
    const vehicle_class = vehicle.vehicle_class;
    const passengers = vehicle.passengers;
    const vehicleCard = {vehicle_name, model, vehicle_class, passengers, type: 'vehicles', id: Date.now() * Math.random()}
    return vehicleCard;
  })
    return Promise.all(unresolvedVehiclePromises);
  }catch(error){
    throw new Error(error.message)
  }
}

export const fetchPlanets = async () => {
  try {
    const url = 'https://swapi.co/api/planets';
    const response = await fetch(url);
    const planetData = await response.json();
    const unresolvedPlanetPromises = planetData.results.map(async planet => {
      const planet_name = planet.name;
      const terrain = planet.terrain;
      const planet_population = planet.population;
      const climate = planet.climate;
      const residents = await fetchResidents(planet.residents);
      const planetCard = {planet_name, terrain, planet_population, climate, residents: residents, type: 'planets', id: Date.now() * Math.random()}
      return planetCard;
    })
    return Promise.all(unresolvedPlanetPromises);
  } catch(error) {
    throw new Error(error.message)
  }
}

const fetchResidents = (residents) => {
  try {
    const unresolvedResidentPromises = residents.map(async resident => {
      const response = await fetch(resident);
      const residentData = await response.json();
      const residentName = residentData.name;
      return residentName
    })
    return Promise.all(unresolvedResidentPromises)
  } catch(error) {
    throw new Error(error.message)
  }
}
