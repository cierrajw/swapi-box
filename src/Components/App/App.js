import React, { Component } from 'react';
import LandingPage from '../LandingPage/LandingPage';
import './App.css';

class App extends Component {
  constructor(){
    super();

    this.state = {
      allCards: [],
      favoriteCards: []
    }
  }
  render() {
    return (
      <div className="App">
        <LandingPage />
      </div>
    );
  }
}

export default App;
