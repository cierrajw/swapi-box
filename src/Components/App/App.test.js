import React from 'react';
import { shallow} from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    wrapper=shallow(<App />);
    mockEvent=jest.fn();
  });

  it.skip('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke displayFilmText on page load', () => {
    wrapper.instance().displayFilmText();
    expect(mockEvent).toHaveBeenCalled();
  });

  it('should update state when displayPeopleCards is invoked', () => {
    const initialState = [];
    const expected = [{name: 'Luke Skywalker', species: 'human'}];
    wrapper.instance().displayPeopleCards();
    expect(wrapper.state('peopleCards')).toEqual(expected);
  });

  it('should call fetch with correct parameters in displayFilmText', () => {
    const url = 'https://swapi.co/api/films/';
    wrapper.instance().displayFilmText();
    expect(window.fetch).toHaveBeenCalledWith(url); 
  });

  it('should update state when displayFilms is invoked', () => {
    const initialState = '';
    const expected='Star Wars movie text';
    wrapper.instance().displayFilms();
    wrapper.setState({filmText: 'Star Wars movie text'});
    expect(wrapper.state('filmText')).toEqual(expected);
  });

  it('should update state when setRedistrict is invoked', () => {
    const initialState = false;
    const expected = true;
    wrapper.instance().setRedirect();
    wrapper.setState({redirect: true});
    expect(wrapper.state('redirect')).toEqual(expected);
  });

});
