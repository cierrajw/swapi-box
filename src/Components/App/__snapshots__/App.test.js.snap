// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="intro-page"
      >
        <h1
          className="swapi-intro-title"
        >
          swapi-box
        </h1>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav"
          onClick={[Function]}
          to="/landingpage"
        >
          <button
            className="intro-swapi-button"
          >
            <img
              alt=""
              height="80"
              src="rebel-alliance.png"
              width="80"
            />
          </button>
        </NavLink>
        <main
          className="main-div"
        >
          <div
            className="crawl-text-div"
          >
            <section
              className="filmtext-content"
            >
              <div
                className="film-text"
              >
                <p />
                <h2 />
                <h5 />
              </div>
            </section>
          </div>
        </main>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="swapi-intro-title"
          >
            swapi-box
          </h1>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            onClick={[Function]}
            to="/landingpage"
          >
            <button
              className="intro-swapi-button"
            >
              <img
                alt=""
                height="80"
                src="rebel-alliance.png"
                width="80"
              />
            </button>
          </NavLink>,
          <main
            className="main-div"
          >
            <div
              className="crawl-text-div"
            >
              <section
                className="filmtext-content"
              >
                <div
                  className="film-text"
                >
                  <p />
                  <h2 />
                  <h5 />
                </div>
              </section>
            </div>
          </main>,
        ],
        "className": "intro-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "swapi-box",
            "className": "swapi-intro-title",
          },
          "ref": null,
          "rendered": "swapi-box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="intro-swapi-button"
            >
              <img
                alt=""
                height="80"
                src="rebel-alliance.png"
                width="80"
              />
            </button>,
            "className": "nav",
            "onClick": [Function],
            "to": "/landingpage",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                height="80"
                src="rebel-alliance.png"
                width="80"
              />,
              "className": "intro-swapi-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "height": "80",
                "src": "rebel-alliance.png",
                "width": "80",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl-text-div"
            >
              <section
                className="filmtext-content"
              >
                <div
                  className="film-text"
                >
                  <p />
                  <h2 />
                  <h5 />
                </div>
              </section>
            </div>,
            "className": "main-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <section
                className="filmtext-content"
              >
                <div
                  className="film-text"
                >
                  <p />
                  <h2 />
                  <h5 />
                </div>
              </section>,
              "className": "crawl-text-div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="film-text"
                >
                  <p />
                  <h2 />
                  <h5 />
                </div>,
                "className": "filmtext-content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p />,
                    <h2 />,
                    <h5 />,
                  ],
                  "className": "film-text",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h5",
                  },
                ],
                "type": "div",
              },
              "type": "section",
            },
            "type": "div",
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="intro-page"
        >
          <h1
            className="swapi-intro-title"
          >
            swapi-box
          </h1>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            onClick={[Function]}
            to="/landingpage"
          >
            <button
              className="intro-swapi-button"
            >
              <img
                alt=""
                height="80"
                src="rebel-alliance.png"
                width="80"
              />
            </button>
          </NavLink>
          <main
            className="main-div"
          >
            <div
              className="crawl-text-div"
            >
              <section
                className="filmtext-content"
              >
                <div
                  className="film-text"
                >
                  <p />
                  <h2 />
                  <h5 />
                </div>
              </section>
            </div>
          </main>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="swapi-intro-title"
            >
              swapi-box
            </h1>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              onClick={[Function]}
              to="/landingpage"
            >
              <button
                className="intro-swapi-button"
              >
                <img
                  alt=""
                  height="80"
                  src="rebel-alliance.png"
                  width="80"
                />
              </button>
            </NavLink>,
            <main
              className="main-div"
            >
              <div
                className="crawl-text-div"
              >
                <section
                  className="filmtext-content"
                >
                  <div
                    className="film-text"
                  >
                    <p />
                    <h2 />
                    <h5 />
                  </div>
                </section>
              </div>
            </main>,
          ],
          "className": "intro-page",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "swapi-box",
              "className": "swapi-intro-title",
            },
            "ref": null,
            "rendered": "swapi-box",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <button
                className="intro-swapi-button"
              >
                <img
                  alt=""
                  height="80"
                  src="rebel-alliance.png"
                  width="80"
                />
              </button>,
              "className": "nav",
              "onClick": [Function],
              "to": "/landingpage",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  height="80"
                  src="rebel-alliance.png"
                  width="80"
                />,
                "className": "intro-swapi-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "height": "80",
                  "src": "rebel-alliance.png",
                  "width": "80",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="crawl-text-div"
              >
                <section
                  className="filmtext-content"
                >
                  <div
                    className="film-text"
                  >
                    <p />
                    <h2 />
                    <h5 />
                  </div>
                </section>
              </div>,
              "className": "main-div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <section
                  className="filmtext-content"
                >
                  <div
                    className="film-text"
                  >
                    <p />
                    <h2 />
                    <h5 />
                  </div>
                </section>,
                "className": "crawl-text-div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="film-text"
                  >
                    <p />
                    <h2 />
                    <h5 />
                  </div>,
                  "className": "filmtext-content",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p />,
                      <h2 />,
                      <h5 />,
                    ],
                    "className": "film-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "h2",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "h5",
                    },
                  ],
                  "type": "div",
                },
                "type": "section",
              },
              "type": "div",
            },
            "type": "main",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
